elide-async/src/test/java/com/yahoo/elide/async/service/AsyncExecutorServiceTest.java

```diff
@@ -58,17 +64,50 @@ public void testAsyncExecutorServiceSet() {
         assertNotNull(service.getRunners());
         assertEquals(60, service.getMaxRunTime());
         assertNotNull(service.getExecutor());
-        assertNotNull(service.getInterruptor());
+        assertNotNull(service.getUpdater());
         assertEquals(asyncQueryDao, service.getAsyncQueryDao());
     }
 
+    //Test for executor hook execution
     @Test
-    public void testExecuteQuery() {
-        AsyncQuery queryObj = mock(AsyncQuery.class);
-        User testUser = mock(User.class);
+    public void testExecuteQueryFail() throws ExecutionException, TimeoutException, InterruptedException {
+
+       AsyncQuery queryObj = mock(AsyncQuery.class);
+       when(queryObj.getAsyncAfterSeconds()).thenReturn(10);
+       service.executeQuery(queryObj, testUser, NO_VERSION);
+       verify(queryObj, times(1)).setStatus(QueryStatus.PROCESSING);
+       verify(queryObj, times(1)).setStatus(QueryStatus.FAILURE);
+    }
 
+    //Test for executor hook execution
+    @Test
+    public void testExecuteQueryComplete() throws InterruptedException {
+
+        AsyncQuery queryObj = mock(AsyncQuery.class);
+        String query = "/group?sort=commonName&fields%5Bgroup%5D=commonName,description";
+        String id = "edc4a871-dff2-4054-804e-d80075cf827d";
+        ElideResponse response = mock(ElideResponse.class);
```

Comment:
I don't understand why we are mocking the response here (or anywhere else).  It looks like this is dead code.  I would remove it.

Reply:
removed